### DISCUSSION TITLE
Sea Ice Colormap

### DISCUSSION THREAD

**Discussion by bnubald** [icenet-ai/icenet#301] [General]
An interesting discussion on sea ice colormap here:

https://github.com/matplotlib/cmocean/issues/93

**Comment by aranas** [comment]
very interesting, thanks for sharing @bnubald!
Which considerations from the thread do you think matter most for the icenet dashboard?

What stood out to me as most relevant
1) not plotting 100% ice as 'true' white (full lightness) may result in NaNs being misinterpreted as ice if users don't actively change default for NaNs
2) non-intuitive color-nature correspondence with black being open water and bluish being 'some ice'.
3) optimizing for best display of nuance and variation (keeping bluish gradient for different levels of ice)

I think 1 is less relevant for an icenet dashboard, since developers can take care of the NaN issue, whereas the 2nd and 3rd point seems relevant for non-scientific end-users who may have pre-conceived color intuition that would interfere with the interpretation of the graph but also will often be interested in the nuance & variability around marginal ice zones.

Is the solution to just stay away from blue colormaps? Lots of operational products I've seen seem to go for purples or rainbow style maps (eg see icysea example below)

![image](https://github.com/user-attachments/assets/95d6f717-cd27-4e3c-97f4-07cab1609d4b)

**Comment by bnubald** [comment]
Hi @aranas 

I agree with the opening points.

Depending on the user:
* If the user is interested in the general trend of sea ice movement over time, then, the non-scientific Blues_r style would be ok.
* If the user is interested in the different thresholds, I think the banded colourmap (like your screenshot), or diverging colourmap would make it much easier to differentiate between them.
* Based on one of the comments, maybe ability to change the data range? i.e., threshold user-specified values would allow more details to show through.

On your points:
1. Yes, it does currently remove nans (land/ocean regions).
2. 3. In addition to above bullet points, maybe continuing the discussion at the source post and suggesting some custom colourmaps for feedback.

The dashboard does current allow for a large selection of colourmaps (around 250, including reversed). Some examples, ignoring the mercator projection:

![image](https://github.com/user-attachments/assets/3ab6e5a9-5537-4279-93df-cf6eb0030a4e)
![image](https://github.com/user-attachments/assets/b093a46c-e963-4cb3-9a47-23d707adebfb)
![image](https://github.com/user-attachments/assets/52cde188-55bf-42ce-bf96-5349d8696718)
![image](https://github.com/user-attachments/assets/be024e5c-9843-4a3c-9740-3f1f30165f2f)
![image](https://github.com/user-attachments/assets/85b50ccc-48d2-41d2-9310-ced63e903c5b)
![image](https://github.com/user-attachments/assets/164783bd-1bcb-4307-af5c-ffe7d1e55b4d)
![image](https://github.com/user-attachments/assets/8bb0763f-81fc-4172-93ce-ab4bf3bc98d6)
![image](https://github.com/user-attachments/assets/d0ada4ba-56f2-43e8-b9c7-333ba341d7d0)
![image](https://github.com/user-attachments/assets/427c8a08-4152-4547-9974-a0e1bcf81304)
![image](https://github.com/user-attachments/assets/172d369b-c50a-4c7d-babd-bed5dcc52db7)
![image](https://github.com/user-attachments/assets/8ccd7bef-1c11-4e92-86f0-de815552e486)

@JimCircadian had done some trials on different colourmaps before.
